uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Validator/Date.php

```diff
@@ -22,7 +24,10 @@ class CRM_HRLeaveAndAbsences_Validator_Date {
   public static function isValid($date, $format = 'Y-m-d')
   {
     $dateTime = DateTime::createFromFormat($format, $date);
-
+    if (!$dateTime && $format != 'YmdHis') {
+      $format = 'YmdHis';
+      $dateTime = DateTime::createFromFormat($format, $date);
+    }
```

Comment:
Agree, if we are going to use CRM_Utils_Date::processDate for each date passed to BAO then it can check only for YmdHis format when validating. Ok, I'll change that.